### PR TITLE
Enhanced logging and client syncing

### PR DESCRIPTION
**Pull request by nmarinel**
We added detail to the console output to diagnose the issues with inconsistency. Also added a syncing mechanism so that all clients start running at the same time regardless of how long it took to deserialize their replay files.

### CODE REVIEW

File: SqlReplay.Console/BucketStats.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace SqlReplay.Console
+{
+    internal class BucketStats
+    {
+        public int timesFellBehind { get; set; }
+        public double minDelayms { get; set; }
+        public double maxDelayms { get; set; }
+        // count of stored procedured called
```
Comment (Balfa) [inline SqlReplay.Console/BucketStats.cs]: Are these comments basically TODO items? or something else? Should they be removed?

File: SqlReplay.Console/Runner.cs [hunk 2]
```diff
@@ -181,14 +195,14 @@ private async Task RunEventsAsync(Run run)
                 {
                     await Task.Delay(timeToDelay);
                 }
-                Console.WriteLine("Starting bucket: " + bucketTimestamp);
+                Console.WriteLine(DateTime.Now + " - Starting bucket: " + bucketTimestamp + $" - timeToDelay = {timeToDelay}");
```
Comment (Balfa) [inline SqlReplay.Console/Runner.cs]: There are a few of these lines that would be a lot easier to read if they didn't mix string interpolation (`$`) with the `+` operator. I'd say stick with one or the other for the extent of the statement. _(my preference would be interpolation :) )_

**Review by Balfa — CHANGES_REQUESTED** [review]

File: SqlReplay.Console/BucketStats.cs (hunk 1 above)
Comment (nmarinel) [reply SqlReplay.Console/BucketStats.cs]: removed the comments and added a note to ADO Feature

File: SqlReplay.Console/Runner.cs (hunk 2 above)
Comment (nmarinel) [reply SqlReplay.Console/Runner.cs]: good call, I cleaned those up

**Review by Balfa — APPROVED** [review]

**Review by Balfa — APPROVED** [review]

**Review by benchamilton — APPROVED** [review]

**Pull request merged**